scripts/agent-trader/agent_trader.py

```
@@ -281,6 +281,120 @@ def _mark_outcome(b):
+            "edge_at_entry": b.get("edge_per_contract"),
+            "marked_at": b.get("marked_at"), "decided": _mark_outcome(b),
+        })
+    return rows
+
+
+def format_positions(rows) -> str:
+    """Deterministic ASCII table of `open_positions` rows. Frames named in the headers."""
+    if not rows:
+        return "no open positions."
+
+    def num(v, sign=False):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `num`

(ANN202)